nerfstudio/cameras/cameras.py

```diff
@@ -676,15 +684,98 @@ def _generate_rays_from_coords(
             directions_stack[..., 1][mask] = torch.masked_select(torch.cos(phi), mask).float()
             directions_stack[..., 2][mask] = torch.masked_select(-torch.cos(theta) * torch.sin(phi), mask).float()
 
+        if CameraType.OMNIDIRECTIONALSTEREO_L.value in cam_types:
+            # Directions calculated similarly to equirectangular
+            mask = (self.camera_type[true_indices] == CameraType.OMNIDIRECTIONALSTEREO_L.value).squeeze(-1)
+            mask = torch.stack([mask, mask, mask], dim=0)
+            theta = -torch.pi * coord_stack[..., 0]
+            phi = torch.pi * (0.5 - coord_stack[..., 1])
+
+            directions_stack[..., 0][mask] = torch.masked_select(-torch.sin(theta) * torch.sin(phi), mask).float()
+            directions_stack[..., 1][mask] = torch.masked_select(torch.cos(phi), mask).float()
+            directions_stack[..., 2][mask] = torch.masked_select(-torch.cos(theta) * torch.sin(phi), mask).float()
+
+            vr_ipd = 0.064  # IPD in meters (note: scale of NeRF must be true to life and can be adjusted with the Blender add-on)
+
+            # find ODS camera position
+            transposedC2W = c2w[0][0].t()
+            ods_cam_position = transposedC2W[3].repeat(c2w.shape[1], 1)
+
+            rotation = c2w[..., :3, :3]
+
+            ods_theta = -torch.pi * ((x - cx) / fx)[0]
+
+            # local axes of ODS camera
+            ods_x_axis = torch.tensor([1, 0, 0]).to("cuda")
```

Comment:
Probably won't make much a difference here since the tensor is small, but it is better to construct the tensor on the target device, rather than move it to the device. Also avoid specifying `cuda` since users may be using cpu.

somethings like:
`ods_x_axis = torch.tensor([1, 0, 0], device=c2w.device)`